### PR TITLE
Pass insertion options to removeRow in List

### LINKED ISSUE #510: Editing children of an expanded row in tree throws Observable error
I am trying to programmatically edit all child rows of a dgrid tree parent when a parent is edited. 

An example would be if a checkbox is checked in the parent row, I want to programmatically check all the checkboxes of the child row:

TreeCheckGroup

Here's a gist that checks all the parent and child rows of 'Europe' when the 'Check Europe' button is clicked : https://gist.github.com/adoe/e8f67a2e1d287b294d49

I see the following console error if the 'Europe' row is expanded during button-click:
message: "Query is out of date, you must observe() the query prior to any data modifications"

I am using dgrid-0.3.6

### PR DESCRIPTION
**Pull request by maier49**
Fixes #510

### CODE REVIEW

File: test/intern/mixins/Tree.js [hunk 1]
```diff
@@ -409,6 +409,34 @@ define([
...
+					grid.collection.putSync({
+						id: '0',
+						value: 'Modified'
+					});
+					grid.collection.putSync({
+						id: '0:0',
+						value: 'Modified',
+						parent: '0'
+					});
+					setTimeout(dfd.callback(function() {
```
Comment (kfranqueiro) [inline test/intern/mixins/Tree.js]: Is there a reason this `setTimeout` needs to be here?  I think this test will work fine so long as you `return` on line 415... (though I mysteriously can't get it to fail for me on FF in either case with or without the fix)

File: test/intern/mixins/Tree.js [hunk 2]
```diff
@@ -409,6 +409,34 @@ define([
...
+						grid.collection.putSync({
+							id: '0:0',
+							value: 'Modified',
+							parent: '0'
+						});
+						assert.strictEqual(grid.row('0').element, grid.domNode.querySelector('.dgrid-row'),
+						'Expected first row to be the first row in the dom');
+					}), 100);
+				})
+			})
```
Comment (kfranqueiro) [inline test/intern/mixins/Tree.js]: Missing semicolons here

File: test/intern/mixins/Tree.js [hunk 3]
```diff
@@ -409,6 +409,34 @@ define([
 					}, null, 'Modification of child should not throw error');
 				});
 			});
+
+			test.test('child modification after parent when expanded', function() {
+				var dfd = this.async(10000);
+				expand(0).then(function() {
```
Comment (kfranqueiro) [inline test/intern/mixins/Tree.js]: Missing spaces between `function` and `(` here and 2 lines up

File: test/intern/mixins/Tree.js (hunk 1 above)
Comment (maier49) [inline test/intern/mixins/Tree.js]: I removed the `setTimeout` and it still works fine, and still fails(in chrome and IE) after removing the fix. I also removed the fourth call to `putSync` as that's not necessary either. Firefox is stumping me at the moment. Manually reproducing the steps from the original ticket without the fix results in the same broken behavior as in chrome, and it also works correctly after adding the fix. Despite that I can't get it to fail in this test case.

File: test/intern/mixins/Tree.js (hunk 2 above)
Comment (maier49) [inline test/intern/mixins/Tree.js]: Fixed

File: test/intern/mixins/Tree.js (hunk 3 above)
Comment (maier49) [inline test/intern/mixins/Tree.js]: Fixed

**Comment by kfranqueiro** [comment]
Merged via https://github.com/SitePen/dgrid/commit/4bf47d13f86bae21f8c6a7a00e38a5a1d3de8c2c (master) and https://github.com/SitePen/dgrid/commit/8351993004185501ea2feddcbf5c2daa7e0cebbd (dev-0.4). Thanks!

**Pull request closed without merging**